File: src/main/java/com/example/backendnourpfe/classes/Demande.java
package com.example.backendnourpfe.classes;


import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;



import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Demande {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idDemande;
    private Date date;
    private String description;
    private  String title;
    private Integer telephoneNumber;

    @Enumerated(EnumType.STRING)
    private StatusDemande statusDemande;
    private  Integer heureTravail;
    private  String demandephoto;

@ManyToOne(fetch = FetchType.EAGER)
@JoinColumn(name = "utilisateur_id")

    private Utilisateur utilisateur;

@ManyToOne(fetch = FetchType.EAGER)

@JoinColumn(name = "service_id")

    private Servicee servicee;

@ManyToOne(fetch = FetchType.EAGER)

@JoinColumn(name = "adresse_id")

    private Adresse adressedemande;
    @JsonIgnoreProperties("demande")
    @OneToMany(mappedBy = "demande", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Postulation> postulations = new ArrayList<>();

    @OneToMany(mappedBy = "demande", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnoreProperties("demande")
    private List<Reservation> reservations = new ArrayList<>();

}

File: src/models/Demande.ts
import { Adresse } from "./Adresse";
import { Postulation } from "./Postulation";
import { Reservation } from "./Reservation";
import { Servicee } from "./Servicee";
import { StatusDemande } from "./StatusDemande";
import { Utilisateur } from "./Utilisateur";

export class Demande {
    idDemande?: number;
    date?: Date;
    description?: string;
    title?: string; 
    telephoneNumber?: number;
    statusDemande?: StatusDemande;
    heureTravail?: number;
    demandephoto?: string;
    utilisateur?: Utilisateur;
    servicee?: Servicee;
    adressedemande?: Adresse;
    postulations?: Postulation[];
    reservation?: Reservation; 
  }
  